client/app/bundles/course/assessment/pages/AssessmentStatistics/LiveFeedbackStatisticsTable.tsx

```diff
@@ -0,0 +1,306 @@
+
+import { getClassnameForLiveFeedbackCell } from './classNameUtils';
+import { getAssessmentStatistics } from './selectors';
+import { getJointGroupsName, translateStatus } from './utils';
+
+const translations = defineMessages({
+  name: {
+    id: 'course.assessment.statistics.name',
+    defaultMessage: 'Name',
+  },
+  group: {
+    id: 'course.assessment.statistics.group',
+    defaultMessage: 'Group',
+  },
+  workflowState: {
+    id: 'course.assessment.statistics.workflowState',
+    defaultMessage: 'Status',
+  },
+  questionIndex: {
+    id: 'course.assessment.statistics.questionIndex',
+    defaultMessage: 'Q{index}',
+  },
+  totalFeedbackCount: {
+    id: 'course.assessment.statistics.totalFeedbackCount',
+    defaultMessage: 'Total',
+  },
+  searchText: {
+    id: 'course.assessment.statistics.searchText',
+    defaultMessage: 'Search by Name or Groups',
+  },
+  filename: {
+    id: 'course.assessment.statistics.filename',
+    defaultMessage: 'Question-level Live Feedback Statistics for {assessment}',
+  },
+  legendLowerUsage: {
+    id: 'course.assessment.statistics.legendLowerUsage',
+    defaultMessage: 'Lower Usage',
+  },
+  legendHigherusage: {
+    id: 'course.assessment.statistics.legendHigherusage',
+    defaultMessage: 'Higher Usage',
+  },
+});
+
+interface Props {
+  includePhantom: boolean;
+  liveFeedbackStatistics: AssessmentLiveFeedbackStatistics[];
+}
+
+const LiveFeedbackStatisticsTable: FC<Props> = (props) => {
+  const { t } = useTranslation();
+  const { courseId } = useParams();
+  const { includePhantom, liveFeedbackStatistics } = props;
+
+  const statistics = useAppSelector(getAssessmentStatistics);
+  const assessment = statistics.assessment;
+
+  const [parsedStatistics, setParsedStatistics] = useState<
+    AssessmentLiveFeedbackStatistics[]
+  >([]);
+  const [upperQuartileFeedbackCount, setUpperQuartileFeedbackCount] =
+    useState<number>(0);
+
+  useEffect(() => {
+    const feedbackCounts = liveFeedbackStatistics
+      .flatMap((s) => s.liveFeedbackCount ?? [])
+      .map((c) => c ?? 0)
+      .filter((c) => c !== 0)
+      .sort((a, b) => a - b);
+    const upperQuartilePercentileIndex = Math.floor(
+      0.75 * (feedbackCounts.length - 1),
+    );
+    const upperQuartilePercentileValue =
+      feedbackCounts[upperQuartilePercentileIndex];
+    setUpperQuartileFeedbackCount(upperQuartilePercentileValue);
+
+    const filteredStats = includePhantom
+      ? liveFeedbackStatistics
+      : liveFeedbackStatistics.filter((s) => !s.courseUser.isPhantom);
+    setParsedStatistics(
+      filteredStats
+        .sort((a, b) => a.courseUser.name.localeCompare(b.courseUser.name))
+        .sort(
+          (a, b) =>
+            (b.liveFeedbackCount?.reduce(
+              (sum, count) => sum + (count || 0),
+              0,
+            ) ?? 0) -
+            (a.liveFeedbackCount?.reduce(
+              (sum, count) => sum + (count || 0),
+              0,
+            ) ?? 0),
```

Comment:
why aren't you precomputing the `totalCount`?  
now the logic becomes inefficient (repeated computations) and hard to parse.

```ts
.sort((a, b) => {
  const byIsPhantom = Number(a.courseUser.isPhantom) - Number(b.courseUser.isPhantom);
  if (byIsPhantom !== 0) return byIsPhantom;

  const byTotalLiveFeedbackCount = b.totalLiveFeedbackCount - a.totalLiveFeedbackCount;
  if (byTotalLiveFeedbackCount !== 0) return byTotalLiveFeedbackCount;

  return a.courseUser.name.localeCompare(b.courseUser.name);
});
```

Reply:
Done